src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5435,6 +5443,13 @@ CalcHeatBalanceInsideSurf( Optional_int_const ZoneToResimulate ) // if passed in
 		}
 	}
 
+	// Calculate heat extract due to additional heat flux source term as the surface boundary condition 
+	for ( SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum ) {
+		if ( Surface( SurfNum ).InsideHeatSourceTermSchedule != 0 ) {
+			QAdditionalHeatSourceInside( SurfNum ) = EnergyPlus::ScheduleManager::GetCurrentScheduleValue( Surface( SurfNum ).InsideHeatSourceTermSchedule );
+		}
+	}
```

Comment:
It's OK to leave it like this, but for an order independent loop like this, it would be nice to simplify it using range-based looping.  You can also eliminate the != 0 check, since in C++, a nonzero value is truthy:

```
for ( auto & thisSurf : Surface ) {
  if ( thisSurf.InsideHeatSourceTermSchedule ) {
    QAdditionalHeatSourceInside( SurfNum ) = EnergyPlus::ScheduleManager::GetCurrentScheduleValue( thisSurf.InsideHeatSourceTermSchedule );
  }
}
```

Consider this a note for later unless you really want to make the change now.

Reply:
I have to keep the index here to get QAdditionalHeatSourceInside( SurfNum ), but it's good to know the preferred structure. I've adopted the nonzero=truthy way.